submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/util/SubmissionFilterSpecification.kt

```diff
@@ -0,0 +1,57 @@
+package ac.uk.ebi.biostd.persistence.util
+
+import ac.uk.ebi.biostd.persistence.model.Submission
+import ac.uk.ebi.biostd.persistence.model.User
+import org.springframework.data.jpa.domain.Specification
+
+class SubmissionFilterSpecification(userId: Long, filter: SubmissionFilter) {
+    var specification: Specification<Submission>
```

Comment:
I would rather avoid mutable specification (`var` and not `val`) as we will be able to change the filter once build, also immutable classes are thread-safe. so we can create a method that handles initialization and make filed and class immutable.

` val specification: Specification<Submission> = createSpecification(userId, filter)`

Reply:
Or maybe using a builder approach

Reply:
Going for the easy-way-out `val`. Let's shift to a generic builder when/if we have to add more filters like this.